src/transaction/TransactionController.ts

```diff
@@ -23,6 +23,25 @@ import {
   query,
 } from '../util';
 
+
+/**
+ * Major issuse attempting to fix with these suggestions is that the tx state stored locally does not
+ * reflect the blockchain. This can result in a number of different app behaviors.
+ * - duplicate transactions
+ * - incorrect transactions state
+ *   - On transaction history we're currently using the gas values that the user sent the transaction with, but in many cases the actual gas spent was less so we are showing misleading informaction.
+ *   - Wrong transaction state: showing failed transaction, when the tx was successful.
+ *   - Wrong transaction state: “Dropped and replaced” shows up in etherscan and our end shows pending.
+ *   - Wrong value being shown on history: showing gas fee instead of the amount swapped.
+ *   - The states of transactions 2 transactions with same nonce are shown wrongly.
```

Comment:
Swap transactions not updating correctly. 
Group by nonce doesn't mean it is the same tx (drop then retry) group by the intent of tx (examine tx params, data field, github link WIP: )